File: lista12a/app/src/main/java/com/example/lista1_2a/MainActivity.java
package com.example.lista1_2a;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private int score;
    private int num1;
    private int num2;
    private int eq;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        score = 0;
        roll();
    }

    private void roll()
    {
        TextView pt = (TextView)findViewById(R.id.points);
        pt.setText("Punkty: " + Integer.toString(score));
        Random rand = new Random();
        num1 = rand.nextInt( 19) + 1;
        num2 = rand.nextInt(19) + 1;
        eq = rand.nextInt(3);
        TextView equation = (TextView)findViewById(R.id.equation);
        equation.setText(Integer.toString(num1) + " ? " + Integer.toString(num2) + " = " + Integer.toString(solve(num1, num2, eq)));
    }

    public void plus(View view) {
        if (eq == 0) {
            Toast.makeText(this, "Dobrze!", Toast.LENGTH_SHORT).show();
            score++;
        }
        else
        {
            Toast.makeText(this, "Źle!", Toast.LENGTH_SHORT).show();
            score--;
        }
        roll();
    }

    public void minus(View view) {
        if (eq == 1) {
            Toast.makeText(this, "Dobrze!", Toast.LENGTH_SHORT).show();
            score++;
        }
        else
        {
            Toast.makeText(this, "Źle!", Toast.LENGTH_SHORT).show();
            score--;
        }
        roll();
    }

    public void multi(View view) {
        if (eq == 2) {
            Toast.makeText(this, "Dobrze!", Toast.LENGTH_SHORT).show();
            score++;
        }
        else
        {
            Toast.makeText(this, "Źle!", Toast.LENGTH_SHORT).show();
            score--;
        }
        roll();
    }

    public void restart(View view) {
        score = 0;
        roll();
    }

    public void nwe_eq(View view) {
        roll();
    }

    private int solve(int num1, int num2, int eq) {
        int result;
        if(eq == 0){
            result = num1+num2;
        }
        else if(eq == 1){
            result = num1-num2;
        }
        else {
            result = num1*num2;
        }
        return result;
    }

}

File: lista12b/app/src/main/java/com/example/lista1_2b/MainActivity.kt
package com.example.lista1_2b

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.lista1_2b.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private var score = 0
    private var num1 = 0
    private var num2 = 0
    private var eq = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        score = 0
        roll()
    }

    private fun roll() {
        binding.points.text = "Punkty:" + score;
        num1 = Random.nextInt(19) + 1
        num2 = Random.nextInt(19) + 1
        eq = Random.nextInt(3)
        binding.equation.text = num1.toString() + " ? " + num2.toString() + " = " + solve(num1, num2, eq)
    }


    operator fun plus(view: View?) {
        if (eq == 0) {
            Toast.makeText(this, R.string.good, Toast.LENGTH_SHORT).show()
            score++
        } else {
            Toast.makeText(this, R.string.bad, Toast.LENGTH_SHORT).show()
            score--
        }
        roll()
    }

    operator fun minus(view: View?) {
        if (eq == 1) {
            Toast.makeText(this, R.string.good, Toast.LENGTH_SHORT).show()
            score++
        } else {
            Toast.makeText(this, R.string.bad, Toast.LENGTH_SHORT).show()
            score--
        }
        roll()
    }

    fun multi(view: View?) {
        if (eq == 2) {
            Toast.makeText(this, R.string.good, Toast.LENGTH_SHORT).show()
            score++
        } else {
            Toast.makeText(this, R.string.bad, Toast.LENGTH_SHORT).show()
            score--
        }
        roll()
    }

    fun restart(view: View?) {
        score = 0
        roll()
    }

    fun nwe_eq(view: View?) {
        roll()
    }


    private fun solve(num1: Int, num2: Int, eq: Int): Int {
        val result: Int

        result = if (eq == 0) {
            num1 + num2
        } else if (eq == 1) {
            num1 - num2
        } else {
            num1 * num2
        }
        return result
    }
}

